src/HIE/Bios/Cradle.hs

```diff
@@ -138,13 +142,20 @@ biosAction _wdir bios fp = do
 -- Works for new-build by invoking `v2-repl` does not support components
 -- yet.
 
-cabalCradle :: FilePath -> Maybe String -> Cradle
-cabalCradle wdir mc =
-  Cradle {
+cabalCradle :: FilePath -> Maybe String -> IO Cradle
+cabalCradle wdir mc = do
+  cabalFiles <- findCabalFiles wdir
+  return Cradle {
       cradleRootDir    = wdir
+    , cradleDependencies = ["cabal.project", "cabal.project.local"] ++ cabalFiles
```

Comment:
are there any overlooked special cases?

Reply:
This will not work if the cabal file is in some sub-directory.
We could recursively search all sub-directories for cabal files, or parse "cabal.project" and "stack.yaml" respectively to get the subdirectories we are interested in.

Reply:
It's close. Ideally stack and cabal would spit this information out themselves.